pkg/controllers/vdb/install_reconcile.go

```diff
@@ -320,6 +320,12 @@ func (d *InstallReconciler) genCreateConfigDirsScript(p *PodFact) string {
 		numCmds++
 	}
 
+	if !p.fileExists[paths.AgentKeyFile] || !p.fileExists[paths.AgentCertFile] {
```

Comment:
We need to skip this if the image doesn't have the agent keys (i.e. it's a NO SSH KEY build or an older image). We probably need to collect that in podfacts in the gather script. Something like 'does the image have the agent keys?'.